### PR TITLE
feat: implemented the UI for complete dashboard and market position d…

### PR DESCRIPTION
**Pull request by temtechie**
@abelbordonado I have made a new PR

### CODE REVIEW

File: src/app/api/user/route.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import { NextResponse, NextRequest } from "next/server";
+import connectDatabase from "@/lib/database";
+import UserSignee from "@/models/user-model";
+
+export async function POST(request: NextRequest) {
+
+    const { signeeWalletAddress } = await request.json();
+
+    await connectDatabase();
```
Comment (abelbordonado) [inline src/app/api/user/route.ts:9]: Normally we need to put this in a global place, not require to be in every route.
At least in node js

File: src/app/api/user/route.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        const newUser = new UserSignee({
+            signeeWalletAddress
+        });
+
+        await newUser.save();
+        return NextResponse.json({ message: "User created successfully" }, { status: 201 }); // Use 201 Created
+
+    } catch (error) {
+        console.error("api/user server Error: ", error);
+        return NextResponse.json({ error: "Something went wrong" }, { status: 500 });
```
Comment (abelbordonado) [inline src/app/api/user/route.ts:27]: We always return the original error (is a security issue but it makes the app much usable and user friendly)

File: src/app/component/common/Tables/active-position-container.tsx [hunk 3]
```diff
@@ -1,34 +1,41 @@
...
+  const [active, setActive] = useState(true);
+  let activePositionsTableData = active
+    ? positionsInfo.slice(0, 2)
+    : positionsInfo;
+  const handleClick = () => {
+    if (active) {
+      setActive(false);
+    } else {
+      setActive(true);
     }
```
Comment (abelbordonado) [inline src/app/component/common/Tables/active-position-container.tsx:18]: I think Eddy made this

`setActive(active);`

File: src/app/component/create-loan/create-loan-form.tsx [hunk 4]
```diff
@@ -1,110 +1,156 @@
...
+                />
+                <select
+                  name=""
+                  id=""
+                  className="bg-slate-100 h-10 cursor-pointer "
+                >
+                  <option value="ETH">USD</option>
+                  <option value="USDC">USDC</option>
+                  <option value="LINK">LINK</option>
+                  <option value="USD">ETH</option>
```
Comment (abelbordonado) [inline src/app/component/create-loan/create-loan-form.tsx:71]: This needs to be from a config object that we can save in `config.ts` or `conts.ts` 
```
ALLOWED_COINS = [{ value: "ETH", icon: "...", token: "ETH" }]
```

Comment (abelbordonado) [reply src/app/component/create-loan/create-loan-form.tsx:71]: After that make this select a input common component <SelectToken  value={} onChange={}/>

File: src/app/component/market/available-positions.tsx [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+                href={tab.href}
+                className="inline-flex items-center justify-center whitespace-nowrap rounded-xl border border-neutral-200 text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 h-10 px-4 py-2 gap-2"
+              >
+                {/* <tab.icon className="w-6 h-6" /> */}
+                <Image src={tab.icon} alt="icon image" height={20} width={20} />
+                {tab.label}
+              </a>
+            </li>
+          ))}
+        </ol>
```
Comment (abelbordonado) [inline src/app/component/market/available-positions.tsx:60]: This need to be in a function wrapper.

When you see such a long CSS is a smell codes that indicates you it need to be abstracted,specially if it is duplicated

File: src/app/component/market/market-position-detail.tsx [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+
+export const MarketPositionDetail = () => {
+  const router = useRouter();
+
+  return (
+    <section className="">
+      <div>
+        <button onClick={() => router.push("/market")}>
+          <IoMdArrowBack className="hover:text-appColor1 h-5 w-5" />
+        </button>
```
Comment (abelbordonado) [inline src/app/component/market/market-position-detail.tsx:15]: We are not using the common buttons here?  each button is having different type of style

File: src/app/api/user/route.ts (hunk 1 above)
Comment (temtechie) [reply src/app/api/user/route.ts:9]: I will find a solution to this comment in next PR, this is a method related to next js 13 + versions

File: src/app/api/user/route.ts (hunk 2 above)
Comment (temtechie) [reply src/app/api/user/route.ts:27]: resolved

File: src/app/component/common/Tables/active-position-container.tsx (hunk 3 above)
Comment (temtechie) [reply src/app/component/common/Tables/active-position-container.tsx:18]: yes

File: src/app/component/create-loan/create-loan-form.tsx (hunk 4 above)
Comment (temtechie) [reply src/app/component/create-loan/create-loan-form.tsx:71]: resolved

File: src/app/component/market/available-positions.tsx (hunk 5 above)
Comment (temtechie) [reply src/app/component/market/available-positions.tsx:60]: I have reduced the redundant css that  are not needed

File: src/app/component/market/market-position-detail.tsx (hunk 6 above)
Comment (temtechie) [reply src/app/component/market/market-position-detail.tsx:15]: I will fix this in next PR

**Pull request merged**